render-test/runner.cpp

```diff
@@ -383,10 +400,90 @@ bool TestRunner::checkRenderTestResults(mbgl::PremultipliedImage&& actualImage,
             return false;
         }
     }
+    // Check gfx metrics
+    for (const auto& expected : metadata.expectedMetrics.gfx) {
+        auto actual = metadata.metrics.gfx.find(expected.first);
+        if (actual == metadata.metrics.gfx.end()) {
+            metadata.errorMessage = "Failed to find gfx probe: " + expected.first;
+            return false;
+        }
+
+        const auto& probeName = expected.first;
+        const auto& expectedValue = expected.second;
+        const auto& actualValue = actual->second;
+        bool failed = false;
+
+        if (expectedValue.numDrawCalls != actualValue.numDrawCalls) {
+            std::stringstream ss;
+            ss << "Number of draw calls at probe\"" << probeName << "\" is " << actualValue.numDrawCalls
+               << ", expected is " << expectedValue.numDrawCalls;
+            metadata.errorMessage += ss.str();
+            failed = true;
+        }
+
+        if (expectedValue.numTextures != actualValue.numTextures) {
+            std::stringstream ss;
+            ss << "Number of textures at probe \"" << probeName << "\" is " << actualValue.numTextures
+               << ", expected is " << expectedValue.numTextures;
+            if (metadata.errorMessage.length()) ss << std::endl;
```

Comment:
should it be checked before adding the new error message?

Reply:
All failed test cases of a probe are reported in a single message. Should this behavior be changed?

Reply:
I mean, should `std::endl` be added first i.e. as a separator between the different probe reports?

Reply:
```
if (!metadata.errorMessage.empty()) ss << std::endl;
ss << "Number of textures at probe \"" << probeName << "\" is " << actualValue.numTextures
               << ", expected is " << expectedValue.numTextures;
```

Reply:
True, metadata is shared between different operations.